app/src/main/java/com/devshish/internship/data/repository/LyricsRepositoryImpl.kt

```diff
@@ -35,10 +36,21 @@ class LyricsRepositoryImpl(
         lyricsDAO.storeSong(roomSong)
     }
 
-    override suspend fun getStoredSongs(): List<SearchSong> =
-        lyricsDAO.getLyrics().map { it.toSearchSong() }
+    override fun getStoredSongs(): Flow<List<SearchSong>> =
+        lyricsDAO.getStoredSongs().map { roomSongList ->
+            roomSongList.map { roomSong ->
+                roomSong.toSearchSong()
+            }
+        }
 
     override fun isSongStored(song: SearchSong): Flow<Boolean> {
         return lyricsDAO.isSongStored(song.title, song.artist)
     }
+
+    override suspend fun deleteSong(song: SearchSong) {
+        val roomSong = lyricsDAO.getStoredSong(song.title, song.artist)
+        roomSong?.let {
+            lyricsDAO.deleteSong(it)
+        }
+    }
```

Comment:
You have lyrics currently loaded on your viewmodel - how about make method that will deleteSong(song: SearchSong, lyrics: String)